### PR TITLE
Employee Report - allow admin to select date range

### PR DESCRIPTION
**Pull request by vaibhav8186**
solved this issue- https://github.com/joshsoftware/the_hunger_terminal/issues/246
Now admin can select date range for downloading employee reports.

### CODE REVIEW

File: app/controllers/reports_controller.rb [hunk 1]
```diff
@@ -27,18 +27,22 @@ def employees_current_month
     @users = User.employee_report(current_user.company_id)
   end
 
-
   def monthly_all_employees
     authorize! :monthly_all_employees, :report_management
-    @users = User.employee_last_month_report(current_user.company_id, Time.current-1.month)
+    today = Date.today
+    from = today - (today.day - 1)
```
Comment (pramodshinde) [inline app/controllers/reports_controller.rb]: No need of calculation here you, can use `today.beginning_of_month` to get month start date

File: app/controllers/reports_controller.rb [hunk 2]
```diff
@@ -27,18 +27,22 @@ def employees_current_month
...
+    @from_date = params[:from] || from.strftime('%Y-%m-%d')
```
Comment (pramodshinde) [inline app/controllers/reports_controller.rb]: Avoid using from and today variable as we are just need a from and to in days not need to consider exact timings. 
Replace above code with 
```ruby
    @from_date = params[:from] || Date.today.beginning_of_month.strftime('%Y-%m-%d')
    @to_date = params[:to] || Date.today.strftime('%Y-%m-%d')
```

File: app/controllers/reports_controller.rb [hunk 3]
```diff
@@ -27,18 +27,22 @@ def employees_current_month
...
+    @to_date = params[:to] || today.strftime('%Y-%m-%d')
```
Comment (pramodshinde) [inline app/controllers/reports_controller.rb]: Why `strftime` is applied on today only, Don't we need it on `params[:to]. strftime `?

File: app/views/orders/order_history.html.haml [hunk 4]
```diff
@@ -1,72 +1,22 @@
...
-              %td= grand_total
-              %td= grand_tax
-              %td= grand_discount
-              %td= grand_extra_charges
-              %td= grand_payable
-              %td
-              %td
+    .row
+      #orderhistory
+        = render 'history_table', orders:@orders
```
Comment (pramodshinde) [inline app/views/orders/order_history.html.haml]: Add new line at the end of the file

File: app/views/orders/order_history.js.haml [hunk 5]
```diff
@@ -0,0 +1 @@
+$('#orderhistory').html("#{j(render 'history_table', orders: @orders)}");
```
Comment (pramodshinde) [inline app/views/orders/order_history.js.haml]: Add new line at the end of the file

File: app/controllers/reports_controller.rb (hunk 3 above)
Comment (vaibhav8186) [reply app/controllers/reports_controller.rb]: from parameters we are getting correct format so , i think don't need to apply strftime

**Pull request merged**